### PR TITLE
Vendor gardener v1.62.2 and Clean up RBAC rules for `dnsrecords`

### PR DESCRIPTION
**Pull request by Kostov6**
**How to categorize this PR?**

/kind enhancement

**What this PR does / why we need it**:
1. Vendor gardener v1.62.2
2. From https://github.com/gardener/gardener/releases/tag/v1.60.0:

> [DEVELOPER] The RBAC for DNSRecords is no longer required by extension controllers (other than DNSRecords) and should be removed where applicable. (https://github.com/gardener/gardener/pull/6973, [@plkokanov](https://github.com/plkokanov))

After updating to gardener/gardener > 1.60 the extension no longer needs the RBAC rules for `dnsrecords`.

**Release note**:

```other dependency
The following dependency is updated:
- github.com/gardener/gardener: v1.59.0 -> v1.62.2
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -4,16 +4,16 @@ go 1.19
...
 	github.com/spf13/cobra v1.4.0
 	github.com/spf13/pflag v1.0.5
 	golang.org/x/tools v0.1.12
-	k8s.io/api v0.25.0
-	k8s.io/apimachinery v0.25.0
+	k8s.io/api v0.25.2
+	k8s.io/apimachinery v0.25.2
 	k8s.io/client-go v11.0.1-0.20190409021438-1a26190bd76a+incompatible
 	k8s.io/code-generator v0.25.0
 	k8s.io/component-base v0.25.0
```
[the remark this answers is not in the document]
Comment (Kostov6) [reply go.mod]: done

**Review by ialidzhikov — DISMISSED** [review]
/lgtm

**Review by ialidzhikov — APPROVED** [review]
/lgtm

**Review by DockToFuture — APPROVED** [review]
/lgtm

**Pull request merged**